specs/DisableCrashReporting.md

```diff
@@ -0,0 +1,120 @@
+{
+    webView.CoreWebView2.ProcessFailed += WebView_ProcessFailed;
+}
+
+void WebView_ProcessFailed(object sender, CoreWebView2ProcessFailedEventArgs e)
+{
+    // When process failed, do custom parsing with dumps
+    string crashDumpFolder = webView.CoreWebView2.Environment.CrashDumpFolderPath;
+    ProcessNewCrashDumps(crashDumpFolder);
+}
+
+```
+## Win32 C++
+```cpp
+void AppWindow::InitializeWebView()
+{
+    auto options = Microsoft::WRL::Make<CoreWebView2EnvironmentOptions>();
+    CHECK_FAILURE(options->put_IsCustomCrashReportingEnabled(TRUE));
+    // ... other option properties
+
+    // ... CreateCoreWebView2EnvironmentWithOptions
+}
+
+void AppWindow::ProcessFailed()
+{
+    // Get dump folder path
+    wil::com_ptr<ICoreWebView2Environment11> environment;
+    wil::unique_cotaskmem_string crashDumpFolder;
+    CHECK_FAILURE(environment->get_CrashDumpFolderPath(&crashDumpFolder));
+	
+	// Register a handler for the ProcessFailed event.
+    CHECK_FAILURE(m_webView->add_ProcessFailed(
+        Callback<ICoreWebView2ProcessFailedEventHandler>(
+            [this, crashDumpFolder](ICoreWebView2* sender, ICoreWebView2ProcessFailedEventArgs* argsRaw)
+                -> HRESULT {
+                // Custom processing
+				ProcessNewCrashDumps(crashDumpFolder);
+                return S_OK;
+            })
+            .Get(),
+        &m_processFailedToken));
+}
+```
+
+# API Details
+```
+interface ICoreWebView2Environment11;
+interface ICoreWebView2EnvironmentOptions3;
+
+/// A continuation of the ICoreWebView2Environment interface for
+/// getting the crash dump folder path
+[uuid(F619312E-0399-4520-B700-30818441785A), object, pointer_default(unique)]
+interface ICoreWebView2Environment11 : ICoreWebView2Environment10 {
+  /// Get the crash dump folder that all CoreWebView2's from this
+  /// environment are using. Whenever a WebView2 process crashes, 
+  /// a crash dump file will be created in the crash dump folder.
+  /// A crash dump format is minidump files, please see
+  /// https://docs.microsoft.com/en-us/windows/win32/debug/minidump-files for 
+  /// detail documentation.
+  /// `ProcessFailed` event are raised everytime a crash happens, minidump files 
+  /// will be written to disk at that time. Developers can rely on `ProcessFailed` and 
+  /// `CrashDumpFolderPath` to customize crash dump experiences.
+  // MSOWNERS: xiaqu@microsoft.com
+  [propget] HRESULT CrashDumpFolderPath([out, retval] LPWSTR* value);
+}
+
+/// Additional options used to create WebView2 Environment.
+[uuid(3FB94506-58AB-4171-9082-E7D683471A48), object, pointer_default(unique)]
+interface ICoreWebView2EnvironmentOptions3 : ICoreWebView2EnvironmentOptions2 {
+
+  /// When `IsCustomCrashReportingEnabled` is set to `TRUE`, Windows won't send crash data to Microsoft endpoint.
+  /// `IsCustomCrashReportingEnabled` is default to be `FALSE`, in this case, WebView respect OS consent. 
```

Comment:
I think this is a misleading property name. This doesn't enable a "custom reporting" mode. It just turns off the default reporting. Maybe `IsCrashReportingEnabled`, default true? With a note that just because crash reporting is enabled, it still may not be reported (based on other policies). This is just one prerequisite for the reports to be uploaded.

Reply:
Please rename:
 * bool CoreWebView2EnvironmentOptions.CustomizeFailureReporting // default false